NOTES.UNIX

```diff
@@ -8,35 +8,53 @@
  Shared libraries and installation in non-standard locations
  -----------------------------------------------------------
 
- Binaries on Unix variants expect to find shared libraries in standard
- locations, such as /usr/lib, /usr/local/lib and some other locations
- configured in the system (for example /etc/ld.so.conf on some systems).
- If the libraries are installed in non-standard locations, binaries
- will not find them and therefore fail to run unless they get a bit of
- help from a defined RPATH or RUNPATH.  This can be applied by adding
- the appropriate linker flags to the configuration command, such as
- this (/usr/local/ssl was the default location for OpenSSL installation
- in versions before 1.1.0):
+ Binaries on some Unix variants expect to find shared libraries in
+ standard locations, such as /usr/lib and possibly /usr/local/lib.  If
+ the libraries are installed in non-standard locations, binaries will not
+ find them and therefore fail to run unless they get a bit of help from
+ a defined runtime shared library search path.
+
+ It is possible to specify additonal directories for the runtime shared
+ library search path when linking a shared library or an application.
+ They are specified with different linking options depending on operating
+ system and versions thereof, and are talked about differently in their
+ respective documentation, where variations of RPATH are the most usual
+ (to be noted: GNU ld.so systems have two such tags, more on that below).
+
+ Possible options to set the runtime shared library search path are:
+
+    -rpath /whatever/path
+    -R /whatever/path
+    -Wl,-rpath,/whatever/path
+
+ OpenSSL's configuration scripts recognise all these options and pass
+ them to the Makefile that they build.
+ (as a matter of fact, it recognises anything starting with '-Wl,' as a
+ linker option, so for example, HP-UX' '-Wl,+b,/whatever/path' would
+ be used correctly)
+
+ Please do not use verbatim directories in your runtime shared library
+ search path!  Some OpenSSL config targets add an extra directory level
+ for multilib installations.  To help with that, the produced Makefile
+ includes the variable LIBRPATH, which is a convenience variable to be
+ used with the runtime shared library search path options, as shown in
+ this example:
 
     $ ./config --prefix=/usr/local/ssl --openssldir=/usr/local/ssl \
-        -Wl,-rpath,/usr/local/ssl/lib
+        '-rpath $(LIBRPATH)'
```

Comment:
I'd say that '-Wl,-rpath,$(LIBRPATH)' is better. Because people like to cut-n-paste and "naked" -rpath won't work for majority of users, a.k.a. Linux users. It would work for Tru64 users though...

Reply:
Okie.